### PR TITLE
feat(efb): flyPadOS v2 🚀

### PR DESCRIPTION
**Pull request by hiaaryan**
Fixes: #4053
Fixes: #4075
Fixes: #4575 

## Important
Remove env file from PR workflow before merge.

## Summary of Changes

flyPadOS v2. Full Layout & UI Rewrite. Special thanks to @Benjozork & @ZigTag! 

- [x] Add New Layout.
- [x] Add Navigraph Support.
- [x] Enable Remaining Settings Options.
- [x] Finishing Touches.
- [x] Remove INOP Pages.
- [x] Adjust All Pages.

## Screenshots (if necessary)

image

# Testing Instructions

Please thoroughly test every page (if you don't have a navigraph subscription don't worry about it).

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Frafire** [comment]
why is this pull not changed to open? they did it all

**Comment by Benjozork** [comment]
@Frafire we are not completely done getting this ready. Please refrain from asking when/why stuff will be ready on pull requests unless you're a maintainer, as it's off topic.

File: src/instruments/src/EFB/ChartsApi/Navigraph.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/ChartsApi/Navigraph.tsx:113]: We should probably have some kind of error handling or at least logging here

File: src/instruments/src/EFB/ChartsApi/Navigraph.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/ChartsApi/Navigraph.tsx:131]: Same

File: src/instruments/src/EFB/ChartsApi/Navigraph.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/ChartsApi/Navigraph.tsx:251]: Same

File: src/instruments/src/EFB/ChartsApi/Navigraph.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/ChartsApi/Navigraph.tsx:265]: Same

File: src/instruments/src/EFB/Dashboard/Widgets/FlightWidget.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Dashboard/Widgets/FlightWidget.tsx:138]: Should be disabled for now

File: src/instruments/src/EFB/Ground/Ground.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Ground/Ground.tsx:172]:
```suggestion
```

File: src/instruments/src/Common/persistence.tsx
Comment (Benjozork) [inline src/instruments/src/Common/persistence.tsx:57]: `NXDataStore::get` has a default value parameter, let's use that instead

File: src/instruments/src/EFB/ChartsApi/ChartFox.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/ChartsApi/ChartFox.tsx:26]: Error handling

File: src/instruments/src/EFB/SimbriefApi/simbriefParser.ts
Comment (Benjozork) [inline src/instruments/src/EFB/SimbriefApi/simbriefParser.ts:17]: Error handling

File: src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleConfig.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleConfig.tsx:150]:
```suggestion
                {parseInt(isDualAxis) === 1 && (
```

File: src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleConfig.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleConfig.tsx:179]:
```suggestion
                {parseInt(isDualAxis) === 0
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by Samueleonard** [comment]
Category: QA Tester
Name: 1Samueleonard1#0001
Date of testing: Current date 06/05/2021
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 1

Changes to observe: new efb layout

Testing technique: Tryout
1. loaded in on runway
2. turned on efb
3. checked through each page - checked each page looked as it should
4. entered simbrief name + pulled fpln 
5. pages checked below

DISPATCH 
         OFP - flight plan still not formatted correctly - have not currently checked other SB formats
         FUEL - functions as before (defueling bug still there?)
GROUND
         All buttons function as before - works
PERFORMANCE 
         Functions as before - works
NAVIGRAPH
         CANNOT TEST NAVIGRAPH FULLY - do not have chart subscription
         however i did scan the barcode and signed in on mobile 
         was logged in - entered ICAO and charts appeared in list
        switched between light/dark theme - demo image flipped
        
        i assume the charts do also work but cannot test, but everything else with the charts works fine

6. Exited the sim and reloaded to check setting persistance
    BEFORE EXITING FOR COMPARISON
image
went back to main menu and closed sim using menu to make sure that settings persist
    AFTER LOADING IN
image

simbrief name did persist but i have never been able to get the sound settings to save, even after closing msfs the 'proper' way - i think this is an issue on my end but i have seen others report it aswell - probably an msfs issue rather than the efb

7. checked the detents page - works as before

Flight notes: 
.settings page not removed from mcdu (useless now?)
.settings page - CIDS empty? no options

Media: n/a

Issues: as above

Steps to achieve the issue n/a

Overall Rating: Good

Conclusions of the testing: a couple of pre-exising issues and questions, but much nicer than the old layout, just needs more testing by others

**Comment by easyjetsimpilot** [comment]
Category: QA Tester
Name: Gavin (easyjetsimpilot)
Date of testing: Current date 06/05/2021
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 1
Changes to observe: New OS v2

Testing technique: Tryout all pages on FlyPad

1. Turned On FlyPad
2. Checked all pages

Dashboard:- All working OK
Dispatch:- OVERVIEW working OK
	- OFP page all ok - however, zoom resets after navigating away from the OFP page
	- FUEL working ok

Ground:- All working ok and calls ground services as expected.
Performance: - All working as expected
Settings:- All worked as expected with the exception to SimBrief username (see below).


3. Checked Navigraph integration
	- Linked with my account no problem. Navigation issue with some charts portrait format as not all the chart is visible. Not possible to navigate/scroll up and down. Charts can be expanded, but this only makes them larger and no more extra information visible. Landscape charts all appear fine. 
	- Not able to view the full list of charts available for an airport as you cannot scroll down the 	list of available charts. 

4. Checked Simbrief Integration :- worked as expected

5. Checked navigation when 'popped out' 
 - An issue here on the Settings tab. You are currently unable to 'pop out' (right Alt + click) the EFB and enter your simbrief username. The only way to do this is into the EFB directly, which causes sim issues with key bindings as you enter your username. Suggest this be fixed as can cause issues. 

Flight notes (When done a full flight): N/A

Media: N/A

Issues (If Any): Navigraph portrait charts not scrollable or movable resulting in half the chart missing. Full list of airport charts not visible. Not able to pop out the flypad and enter the simbrief username on the settings page. 

Overall Rating: To be improved
Conclusions of the testing: Only 3 issues I could find, but all I feel are integral to the PR. Especially the full charts being able to be viewed. 

Thank you so much for your work on this PR. Looking forward to its release. Hope this has helped.

**Comment by ExampleWasTaken** [comment]
Category: QA Testers
Name: `ExampleWasTaken#0886`
Date of testing: 6/5/2021
Version of the sim: 1.15.10.0
PR Tested: #4026 
Tier of testing: 1
Changes to observe: flyPadOS-v2

Testing technique:
1. Tested dashboard page
2. Tested dispatch page
3. Tested ground page
4. Tested performance page
5. Tested settings page



Media: 
Microsoft Flight Simulator Screenshot 2021 05 06 - 22 54 47 67

[Video of the performance](https://streamable.com/t8v6je)


Issues (If Any): 
1. Dashboard displays `[object Object]` when OFP contains no flight number
2. "PTU audible in cockpit" and volume slider setting is not persistent
3. Terrible performance of the EFB

Steps to achieve the issues:
1. Load OFP with no flight number specified
2. Disable and set sliders to 0 `->` reload sim 
3. The current performance of the EFB needs to be improved.

**My setup**
I9-10900K
NVIDIA 2070 super

Overall Rating: To be improved
Conclusions of the testing: Issue 1 & 2 are minor but the performance should definitely be improved.

### A little suggestion
Use the primary color of the sim for all buttons on the settings page not only the sliders. :)

**Comment by darkfury334** [comment]
Name: DarkFury#4936
Date of testing: Current date 06/05/21
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 1
Changes to observe:
EFb New OS v2

Testing technique: Tryout

Steps you've made to test the Pull Request:
1. Spawn in , turn on efb
2. Check all pages and setting work

Dashboard: all working except for link MCDU
Dispatch: OFP ok, defueling bug still exists on fueling page.
Performance: all working
Navigraph: working but with issues. Needs work
Settings: few issues

Flight notes (When done a full flight):N/A

Media: N/A
Microsoft Flight Simulator Screenshot 2021 05 06 - 22 44 15 93
Microsoft Flight Simulator Screenshot 2021 05 06 - 22 47 56 04

Issues (If Any): 
Link mdcu greyed out
defueling bug still exists. entered lower fuel that current and use real time, started defueling.
CIDS has no options
Navigraph unable to scroll, missing most of information from charts. Issue with some charts in portrait format as not all the chart is visible. Unable
to navigate/scroll up and down. Charts can be made larger and no extra information  is visible.
Not able to view the full list of charts available for an airport as you cannot scroll down the list of available charts.
Slight performance hit when using EFB

Overall Rating: To Be Improved
Conclusions of the testing: Great PR but with a number of issues found which i think should be definitely improved especially the Navigraph page.
Pre-existing issue are still exist in PR.

**Comment by ExampleWasTaken** [comment]
> 
> 
> LGTM

Make sure to use the QA template so the author can retrace the steps you've taken to test the PR.

**Comment by ElonKerman** [comment]
Name: Elon Kerman#9253
Date of testing: Current date 06/05/21
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 1
Changes to observe:
EFb New OS v2

Testing technique: Spawned on runway

Steps you've made to test the Pull Request:

Spawn in , turn on efb
Check all pages and setting work
Dashboard: all working except for link MCDU
Dispatch: OFP ok, defueling bug still exists on fueling page.
Performance: all working


Flight notes (When done a full flight): Game crashed but IDK if that is a issue with plane or my PC

Media: N/A

Overall Rating: LGTM

**Comment by Mack674** [comment]
Category: QA Tester
Name: Mack674#2274
Date of testing: Current date 07.05.2021
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 1
Changes to observe:
- FlyPad OS v2
- Reworked EFB Pages

Testing technique: Tryout all FlyPad pages and new UI
Steps you've made to test the Pull Request:
1. Spawned at gate at EGLL
2. Turned on batteries and external power
3. Turned on EFB
4. Checked all FlyPad pages:
- Dashboard
- Dispatch (Overview, OFP, Fuel)
- Ground
- Performance (TOD)
- Charts
- Settings


Flight notes (When done a full flight): -

Media: 

EFB_main
EFB_SimbriefFP
EFB_DispatchOverviev
EFB_DispatchOFP
EFB_DispatchFuel
EFB_Ground
EFB_PerformanceTOD
EFB_Charts
EFB_Settings

Issues (If Any): No issues

Overall Rating: Very good

Conclusions of the testing: Very good update. New layout is looking much better than older. Unfortunetaly I haven't got a Navigraph Charts subscription, so I can't check that page, but everything else works.

**Comment by oim1** [comment]
Category: QA Tester
Name: oim#0001
Date of testing: 07/05/2021
Version of the sim: 1.15.10.0
PR Tested: #4026
Tier of testing: 2
Changes to observe:
New EFB layout, Navigraph charts integration, Enabling remaining settings, Removing INOP pages.

Testing technique: Full Flight
NAVIGRAPH:
Tested integration. 
Pulled up some charts from different airports. 

General:
Checked for inop pages. 
Checked the settings page. 
Looked at OFP and Fuel page. 
Tested TOD calculator. 
Tested everything on ground page. 
Tested simbrief integration. 

Flight notes: 
On the navigraph charts section, the lack of a scroll bar when selecting charts was confusing at first. 
Having to input the airport every time I went back into the charts page was also annoying after a while. 
No scroll bar on the settings page as well. Gets confusing. 

Media:
chartscroll
settingsscroll

Issues:
Occassional frame drops when using the EFB. 

Overall Rating: To be improved

Conclusions of the testing: 
Many features seem good. It's working as intended. This PR also fixes a few issues. Not sure if it's my system but the EFB causes my frames to drop occasionally 
when looking at it. The lack of scroll bars in a few places is annoying. I've shown the locations in the media attached. Overall it seems to be working fine. Just a few things
need to be looked at. 

Note that scrolling does work on the settings and charts page but you have to find the scrolling location due to the lack of a scroll bar.

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 7/5/21
Version of the sim: 1.15.10.0
PR Tested: #4026 
Tier of testing: 1
Changes to observe:
EFB

Testing technique: Loaded in and looked
Steps you've made to test the Pull Request:
1. Loaded in
2. Used the SimBrief and navigraph
3. Tried other features

Flight notes (When done a full flight):

Media: Insert screenshots, recordings, or anything relevant to the review here.
image
image

Issues (If Any):

- When importing a simbrief plan without a callsign it says [object Object]
- sound settings do not save and are hard to put back in the 0 spot, I think there should be some sort of "snap" which happens very close to the middle

Steps to achieve the issue

Overall Rating: To be improved
Conclusions of the testing: Good work, efb looks very good but there a few issues

**Comment by Snapmatics** [comment]
## Quality Assurance | Development Team

- **Discord:** `Charalampos#0001`
- **Object of Testing:** `feat(efb): flyPadOS v2 🚀#4026`
- **Tier of Testing:** `1`
- **Date:** `7/06/2021`

## Testing Process:

- Spawned at a Gate in Athens (LGAV) and turned ON the flyPadOS v2 for further testing after the big changes.

## Issues
- Map is still using the white theme and not the dark theme. (1)
- Keybinds still interact with the plane when you start typing in a text field. (2)
- Font size is drastically small in a lot of sections of the flyPadOS. (3)
- Charts section will need a `+` `-` button to be able to zoom-in/out the charts. You can't really see their text most of the time. (4)
- Icons in the Charts section can be dragged when you scroll a chart. (5)
- A lot of pages will need to cache their content as you will lose all the real feeling. (You can't check the weather or do anything on the flyPadOS and do other things at the same time). (6)

## Images
Here're the images attached with the issues above.

### Map | White Theme
image

### Keybinds | Still Interact with the Simulator
https://user-images.githubusercontent.com/43104362/120939470-f1877780-c720-11eb-9321-39a8fdb3b724.mp4

### Font Size | Small in a lot of Sections
image
image
image

### Charts [1] | Need `+` `-` Buttons
image

### Charts [2] | Icons move when you drag
https://user-images.githubusercontent.com/43104362/120939663-bc2f5980-c721-11eb-88fb-693e6e739269.mp4

### Cache | Multitasking Issue
https://user-images.githubusercontent.com/43104362/120939798-6c9d5d80-c722-11eb-921d-71c2ba06e789.mp4

**Comment by Cryptic56** [comment]
Quality Assurance Tester/Trainee Report

Discord: (Only use if Discord name is significantly different to GH Name - delete line if not applicable)
Object of testing: Cryptic#4731
Tier of Testing: 1
Date: 20/06/2021

Testing Process: Spawned at YPAD
Loaded up EFB
Imported flight plan (YPAD-YMML)
checked all pages and functions of EFB

Negatives: Imported flight plan with no callsign and it says Object object.
When looking at charts, exiting and then going back to the charts, you have to enter the airport ICAO again
Not very important but when using the landing performance calculator, exiting and going back to the performance tab, it goes back to the TOD calculator instead of the last page that was being used there.


Testing Results: Passed

Conclusions: There were a few issues but generally every function works well. Would it be possible to add your current position on the Navigraph charts that are eligible for this, like in the Navigraph app and WT aircraft?
Capture

**Comment by Benjozork** [comment]
I will look into the graphical issues on the map controls and settings, it was fixed locally, don't know why the CI build didn't have it fixed.

**Comment by Awemeter** [comment]
Quality Assurance Tester

Discord: Awemeter#0321
Object of testing: #4026
Tier of Testing: 1
Date: 6/20/2021

Testing Process: Spawned in at EIDW and tested all functions of efb
Negatives: Moving maps should be added, 122.8 is not an active controller and should be removed, the battery on the iPad should be green when charging. The Map icons are black so I can not see them. The PTU switch in EFB does not work.
Testing Results: Not Passed

Conclusions: Not that many bugs but a lot of necessary features are missing
Media:

**Comment by Benjozork** [comment]
@Awemeter moving maps is not a feature of the PR, and neither is battery charging...

**Comment by Benjozork** [comment]
Also, please be more precise in how the switch doesn't work, and provide screenshots...

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #4026
Tier of Testing: 1
Date: 21.6.2021

Testing Process: 
C&D at EDDF
Simbrief flightplan EDDF-EDDM
Checking each page and subpage
Dashboard
- Simbrief import ==> OK [1]
Dispatch
- OFP ==> OK (EZY format)
- Fuel ==> OK (Defueling, Fueling)
Ground
- Check all button functions
Performance
- TOD: Ground Speed - see negatives ==> otherwise OK
- Landing: see negatives ==> otherwise OK
- Navigraph; ==> see negatives
- ATC ==> OK
Settings ==> OK

Negatives: 
- General: 
  - Last subpage is not remembered - when changing to another page and back the first subpage is visible and not the last used [3] (this is annoying when for example on Landing performance jumping to Dashboard or OFP to lookup values)
- Direct after starting tablet: Map icons transparent [0]
- Link MCDU - not working / unclear what it should do
- Ground Speed - Min Alt looks funny - can't edit (probably not meant to be editable) [2]
- Landing: Metar does not fetch Wind Magnitude [4]
- Landing: Clear does not work [4]
- Navigraph: 
  - Insufficient .env file - unclear how to access Navigraph subscription (self compiled) - Works when run via PR artefact [5]
  - looses selection when changing to other page and back 
  - List of charts - can't scroll with mouse wheel
- ATC: 
  - ATIS font too small [6]
  - no ATIS when using PR artefact / worked with self-compiled build [6][7]

Testing Results: Not Passed

Conclusions: 
Great feature but still too many small issues imho. 
Also there seem to be differences when compiled locally vs. PR build

Media:
[0] Map icons transparent
image
[1] Simbrief OFP
image
image
[2]
image
[3]
https://user-images.githubusercontent.com/16833201/122691149-1c4bf280-d22e-11eb-9b5e-6e186755424b.mp4
[4]
https://user-images.githubusercontent.com/16833201/122691409-c8daa400-d22f-11eb-9188-9e3b100f1eb4.mp4
[5] Navigraph
image
[6] ATC
image
[7]
image

**Comment by Benjozork** [comment]
> Insufficient .env file - unclear how to access Navigraph subscription (self compiled) - Works when run via PR artefact [5]

This is expected and working as intended.

**Comment by Benjozork** [comment]
The latest commit should fix the map panel and settings issue.

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #4026 
Tier of Testing: 2
Date: 21/06/2021

**Testing Process:**
1. Loaded C&D an setup Settings page like I want it to, calibrated throttles -> Check issues
2. Checked dispatch page and refueled -> Check issues
3. Tested all ground services and pushback
4. Checked ATC page on this VATSIM flight, I do have some improvement suggestions, but will do that in discord
5. I do not have a navigraph charts sub
6. Played with ToD and LDG Calculators inflight


**Negatives:** 
- PTU toggle in settings page does not move, it just changes color
- Something was wrong with the Dispatch page. I had no option to load my simbrief FPL. Username was set in the Settings menu. Consequently i also couldnt test OFP page
- I also undestand the last subpage not being remembered may be a little annoying

**Testing Results:** Not Passed due to the dispatch page problem, other than that i think its good to go

Conclusions: (Notes made, the positives in the PR, and anything extra you would like to mention - Delete Line if Nothing Extra is to be mentioned
Media: 
image
image
image

**Comment by easyjetsimpilot** [comment]
Quality Assurance Tester

Discord: easyjetsimpilot
Object of testing: #4026 
Tier of Testing: 2
Date: 21/06/21

Testing Process: Full flight which tested all pages and functions of the EFB OS v.2 (with Navigraph). All functions worked as expected. 

Negatives: 
1. OFP zoom/page not maintained when navigating away from the OFP page
2. Navigraph charts & selected airport not maintained when navigating away from the OFP page. Personally, I see this a a major issue. As you have to pop-out the EFB to type in the ICAO code (due to keybindings), this is time consuming if has to be done every time. 
3. VATSIM page - not all available controllers were shown as available on the page when they were online (according to VPilot) - *image to follow)
Testing Results: Not Passed

Conclusions: Overall, everything works as it should - however, the 3 items above I feel need investigating - particularly the ability to maintain the pages of the VATSIM charts if navigating away from the page. Thanks for all the continued work.

**Comment by ZeroKaa** [comment]
> **Quality Assurance Tester**
> 
> Object of testing: #4026
> Tier of Testing: 2
> Date: 21/06/2021
> 
> **Testing Process:**
> 
> 1. Loaded C&D an setup Settings page like I want it to, calibrated throttles -> Check issues
> 2. Checked dispatch page and refueled -> Check issues
> 3. Tested all ground services and pushback
> 4. Checked ATC page on this VATSIM flight
> 5. I do not have a navigraph charts sub
> 6. Played with ToD and LDG Calculators inflight
> 
> **Negatives:**
> 
> * PTU toggle in settings page does not move, it just changes color
> * Something was wrong with the Dispatch page. I had no option to load my simbrief FPL. Username was set in the Settings menu. Consequently i also couldnt test OFP page
> * I also undestand the last subpage not being remembered may be a little annoying
> * ATC page does not update, when staying on it. But changing to another page and back to ATC, frequencies will update
> * ATC page didnt give me the TWR, GND and ATIS of LOWW although being within 50nm of the airport on approach
> 
> **Testing Results:** Not Passed due to the dispatch page problem, other than that i think its good to go
> 
> Conclusions: (Notes made, the positives in the PR, and anything extra you would like to mention - Delete Line if Nothing Extra is to be mentioned
> Media:
> image
> image
> image
> image

Hello. The ATC page is refreshing every two minutes. Wish is also the refresh rate of data coming from vatsim/Ivao.

**Comment by frankkopp** [comment]
Tested last commit for the map icons - **OK now**
image

**Comment by St54Kevin** [comment]
> 
> Hello. The ATC page is refreshing every two minutes. Wish is also the refresh rate of data coming from vatsim/Ivao.

My page did not update at all when coming into my Destination airport. So that's like 20min without an update. Even changing pages did not do anything.

**Comment by ZeroKaa** [comment]
> > Hello. The ATC page is refreshing every two minutes. Wish is also the refresh rate of data coming from vatsim/Ivao.
> 
> My page did not update at all when coming into my Destination airport. So that's like 20min without an update. Even changing pages did not do anything.

Ok Thanks. I'll do some extra tests because I cannot reproduce it. I'll plan a new flight

**Comment by ZeroKaa** [comment]
> > Hello. The ATC page is refreshing every two minutes. Wish is also the refresh rate of data coming from vatsim/Ivao.
> 
> My page did not update at all when coming into my Destination airport. So that's like 20min without an update. Even changing pages did not do anything.

@St54Kevin . I spot the issue and fixed it. PR still need to be merge
I also increased the font-size

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #4026
Tier of Testing: 1
Date: 1/07/2021

Testing Process:
- Spawned C&D at EDDP
- Start up AC and EFB
- Checked Dashboard page and Simbrief sync
- Checked Dispatch Overview, OFP and Fuel page
- Checked Ground page
- Took off
- Checked Performance page (TOD and Landing)
- Checked Navigation & Charts page (QR Code only)
- Checked VATSIM page (i dont use VATSIM)
- Checked Settings page
- Checked Detents page

Negatives: 
- Map plane flickers
- OFP dont remembers scroll position and zoom level
- Fuel page plane looks a little pressed (close to the Refuel box), maybe put the fuel time and the plane a litte down
- Fuel page fontsizes are inconsistent (left outer tank value, right outer tank value) (fig1)
- Landing page fontsizes are inconsistent/to small (clear, low, medium, manual) (fig2)
- I dont like the grey hairlines between each line

Testing Results: (Not) Passed
Conclusions: Works, just minor fixes
Media:


fig1


fig2

Hope this helps :-)

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #4026 
Tier of Testing: 2
Date: 12/07/2021

**Testing Process:**
1. Full VATSIM EGLL-EDDF flight
2. Went through all the settings and set everything up to my liking
3. Loaded simbrief flight plan, refueled via EFB
4. Tested every ground service
5. caculated relanding performance
6. Used ATC tab to tune frequency
7. EFB pushback did not work
8. Full flight with monitoring ATC tab and vPilot Controllers
9. Unfortunately all controllers went offline right after my departure, therefor I wrote this qa airborne (nothing more to test)

**Negatives:** 
- Sometimes the EFB doesnt take my click when switching pages. I need to click several times to get them to open. But its just from time to time but noticeable
- After calling the baggage, shutting it off didnt work, button was blue but services were stuck at the plane, the door even stayed open with ENG running(see media), I think this was causing my EFB pushback to not work. Had to use pushback toolbar for both problems. I literally did a full flight this evening with master and it was working fine there

**Testing Results:** Not Passed, due to the ground services problems. I will recheck this after landing with a non addon sim, maybe the pushbar toolbar was the problem, which would be a sad because I prefere using it above the EFB function

Media:
image
image
image

**Comment by St54Kevin** [comment]
Just tested with the same sim on a different airport (EDDW) and ground services just worked fine. Maybe that was due to the EGLL scenery.

**Pull request merged**
